lib/tests/formulae.rb

```diff
@@ -423,6 +423,7 @@ def bottle_reinstall_formula(formula, new_formula, args:)
         end
 
         ENV["HOMEBREW_BOTTLE_SUDO_PURGE"] = "1" if MacOS.version >= :catalina
+        ENV["HOMEBREW_NO_PATCHELF_RB_WRITE"] = "1" unless ENV["HOMEBREW_PATCHELF_RB_WRITE"].present?
```

Comment:
Could this be a `with_env(HOMEBREW_PATCHELF_RB_WRITE: "1") do` around `test "brew", "bottle", *bottle_merge_args`?

Reply:
Done! Using the `env` argument of `test` rather than `with_env`.